ngraph/src/ngraph/ngraph.hpp

```diff
@@ -91,8 +91,23 @@ namespace ngraph
 #include "ngraph/node.hpp"
 #include "ngraph/ops.hpp"
 #include "ngraph/partial_shape.hpp"
+#include "ngraph/rt_info.hpp"
 #include "ngraph/shape.hpp"
 #include "ngraph/shape_util.hpp"
 #include "ngraph/specialize_function.hpp"
 #include "ngraph/type.hpp"
 #include "ngraph/type/element_type.hpp"
+#include "ngraph/variant.hpp"
+
+// nGraph opsets
+#include "ngraph/opsets/opset.hpp"
```

Comment:
I understand that these changes are to simplify the ngraph usage from the user's perspective. On the other hand I think all those includes added below bloat the overload set quite a lot in any translation unit that includes `ngraph.hpp`.

Can we consider including only the most recent opset here instead of 1-4?

Reply:
@tomdol Unfortunately, some places can use operations from opset1 or 2 (like ngraph::opset1::SomeOp). If we use include only last opset in this file, each release we will affect some users because they should add additional header files or migrate to new opset.

Maybe in this case we shouldn't include opsetX.hpp headers to this file, but I think it doesn't affect the final binary size or something like that... It just add access to more operations. Do you have any thoughts about it?

I extend this file in order to make backward compatibility support easier (if we change location of some files for example).

Reply:
I will remove opsetX.hpp headers for main nGraph header.

Reply:
That was just some considerations. The final binary size will not be affected because the compile will strip unused stuff. An extender overload set might affect the compilation time. I don't have any numbers though, so for users convenience let's leave it as is.

Reply:
Yes I agree, I also thought about compilation time! I removed all opsets files from main header. If user wants to use the special opset he can use method `get_opsetX()` or include a specific header file.